src/consts.h

```diff
@@ -44,13 +47,24 @@ typedef enum {
 } TS_AGG_TYPES_T;
 
 /* Series struct options */
-#define SERIES_OPT_UNCOMPRESSED 0x1
+#define SERIES_OPT_UNCOMPRESSED (1 << 0)
+#define SERIES_OPT_OUT_OF_ORDER (1 << 1)
 
 /* Chunk enum */
 typedef enum {
   CR_OK = 0,    // RM_OK
   CR_ERR = 1,   // RM_ERR
-  CR_END = 2  
+  CR_END = 2,   // END_OF_CHUNK
+  CR_OCCUPIED = 3,
+  CR_DEL_FAIL = 4,
 } ChunkResult;
 
+typedef enum {
+  UPSERT_NOT_ADD = 0,
+  UPSERT_ADD = 1,
```

Comment:
This will allow for an additional argument in TS.ADD.
I don't think the user should check before. The same way we allow to add for a series w/o checking whether it exists or not.

Reply:
this is not part of the requirements, lets implement the minimum we need at this moment which is doing any change in the history. if we want to add some other semantic we can do that with using a different command or flags.